blocks/core.py

```diff
@@ -239,29 +232,32 @@ def partitioned(
     except ImportError:
         raise ImportError("Partitioned requires dask[dataframe] to be installed")
 
-    grouped = _collect(path, cgroups, rgroups, filesystem)
+    grouped = _collect(path, cgroups, rgroups, filesystem, tmpdir)
     blocks = []
 
     @dask.delayed()
     def merged(rgroup):
         frames = []
         for cgroup in grouped:
-            datafile = next(
-                d for d in grouped[cgroup] if os.path.basename(d.path) == rgroup
-            )
+            p = next(p for p in grouped[cgroup] if os.path.basename(p) == rgroup)
             args = read_args.copy()
             if cgroup in cgroup_args:
                 args.update(cgroup_args[cgroup])
-            frames.append(read_df(datafile, **args))
+            frames.append(read_df(p, **args))
         return _merge_all(frames, merge=merge)
 
     for rgroup in _shared_rgroups(grouped):
         blocks.append(merged(rgroup))
     return dd.from_delayed(blocks)
 
 
+@with_function_tmpdir
 def place(
```

Comment:
seems pretty similar to https://filesystem-spec.readthedocs.io/en/latest/features.html#remote-write-caching ?

Reply:
Yeah there's definitely a lot of overlap, but this does handle a copy of other details including using a local path (not a file handle) which is more compatible with pandas functions and infering the pandas function to use based on file extension